### PR TITLE
Feature/user like

### PR DESCRIPTION
**Pull request by Mikazian**
## Description

BACKEND : 

- Create route to get interactions,

FRONTEND : 

- Create route to display the interactions sent and received,
- Create route to display a error page,
- Add to Navbar the route to navigate in the interactions component,
- Add a function onClick to the Button component,
- Create Error component to display the error statut,
- Create a Layout to display the interactions,
- Create a card to display the interactions and the users,

SHARED : 

- Create two interface for interaction sent and received

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [x] Frontend
- [x] Backend
- [ ] Other: .....

## General checklist:

- [x] My code follows the style guidelines of this project
- [x] I ran `bun run format`/`bun run lint` before commit
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation where needed
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] I have updated `develop` and merged to my branch before submitting pull request
- [x] My pull request generate no conflicts with `develop` branch
- [x] I requested code review from other team members

## Frontend checklist:

- [x] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [x] My `Javascript` generate no new console errors
- [x] I tested my code cross browsers
- [x] My slice is pixel perfect for both desktop and mobile according to design
- [x] I conducted basic QA to assure all features are working
- [x] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [x] I tested admin by manually adding content from zero
- [x] I created easy to use admin experience which is self-explanatory
- [x] I added description to admin fields in hard-to-understand areas
- [x] I followed guidelines for naming `Typescript` variables
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by greegouze — DISMISSED** [review]

**Review by greegouze — DISMISSED** [review]

**Review by greegouze — DISMISSED** [review]

File: packages/frontend/web/src/contexts/UsersInteractionsContext.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    [isVisible],
+  );
+
+  // Fetch the users the user logged in have interacted with
+  useEffect(() => {
+    const controller = new AbortController();
+    const signal = controller.signal;
+
+    fetchInteractions({
+      userId: Number(userId),
```
Comment (YDrogen) [inline packages/frontend/web/src/contexts/UsersInteractionsContext.tsx:103]: Pourquoi t'envoie le userId ? C'est déjà dans les cookies

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/contexts/UsersInteractionsContext.tsx (hunk 1 above)
Comment (Mikazian) [reply packages/frontend/web/src/contexts/UsersInteractionsContext.tsx]: J'ai retiré les userId uniquement sur cette PR. Je ferai un fix pour retirer les autres userId

**Review by greegouze — DISMISSED** [review]

File: packages/frontend/web/src/pages/Home.tsx [hunk 2]
```diff
@@ -14,25 +14,53 @@ export default function Home() {
...
+            </SidebarLayout>
+          ) : undefined}
+
+          {/* Display messages in all page when the width is under to 1024px */}
+          {window.innerWidth < 1024 ? (
+            <SidebarLayout isVisible={isVisibleFilter} isBorderRight>
+              <SidebarLayout isVisible={isVisibleFilter} isBorderLeft>
+                {`Messages`}
+              </SidebarLayout>
+            </SidebarLayout>
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/Home.tsx:46]: Pourquoi t'as une Sidebar dans une sidebar ?

Comment (Mikazian) [reply packages/frontend/web/src/pages/Home.tsx]: corrigé

**Review by greegouze — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Pull request merged**